File: cpp/GroupAnagrams.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>


using namespace std;
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> h;

        for(int i = 0; i < strs.size(); i++){
            string s = strs[i];
            sort(s.begin(), s.end());
            h[s].push_back(strs[i]);
        }
        vector<vector<string>> res;
        
        for(auto p : h){
            res.push_back(p.second);
        }
        return res;
    }
};

File: python/GroupAnagrams.py
from typing import List

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        h = {}

        # Anagrams will be equal if we sort them, considering that we can map group anagrams to the same key in hashMap
        for w in strs:
            sortedW = "".join(sorted(w))
            if sortedW not in h:
                h[sortedW] = [w]
            else:
                h[sortedW].append(w)
        res = []

        for value in h.values():
            res.append(value)
        
        return res